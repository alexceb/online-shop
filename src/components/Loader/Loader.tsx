import * as React from 'react';
import * as styles from './Loader.scss';

export const Loader: React.FC = () => {
  return (
    <div className={styles.loaderContainer}>
      <div className={styles.loader}></div>
    </div>
  )
}
