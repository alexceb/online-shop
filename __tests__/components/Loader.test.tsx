import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Loader } from '../../src/components/Loader/Loader';

describe('Loader component', () => {
  describe('render()', () => {
    test('Renders <Loader /> correctly', () => {
      const wrapper = shallow(<Loader />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('Contains certain elements', () => {
      const wrapper = shallow(<Loader />);

      expect(wrapper.find('.loaderContainer')).toHaveLength(1);
    })
  });
});
