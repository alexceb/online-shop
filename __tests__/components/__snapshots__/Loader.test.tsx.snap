// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component render() Renders <Loader /> correctly 1`] = `
<div
  className="loaderContainer"
>
  <div
    className="loader"
  />
</div>
`;
